components/access-management/kfam/api_default.go

```diff
@@ -114,6 +119,33 @@ func (c *KfamV1Alpha1Client) CreateBinding(w http.ResponseWriter, r *http.Reques
 	}
 	// check permission before create binding
 	useremail := c.getUserEmail(r.Header)
+
+	// --------------------- Added by souheil.yazji --------------------------- //
+	// get profile from namespace
+	profileName, err := c.profileClient.Get(binding.ReferredNamespace, metav1.GetOptions{})
+	if err != nil {
+		IncRequestErrorCounter("Permissions check error", "", action, r.URL.Path,
+			SEVERITY_MAJOR)
+		writeResponse(w, []byte(err.Error()))
+		w.WriteHeader(http.StatusForbidden)
+		return
+	}
+	// check value of label
+	internal_fdi_bucket, err := strconv.ParseBool(profileName.Labels[EXISTS_INTERNAL_BLOB_STORAGE])
+	if err != nil {
+		IncRequestErrorCounter("Label parse failure", "", action, r.URL.Path,
+			SEVERITY_MAJOR)
+		writeResponse(w, []byte(err.Error()))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	// respond forbidden if label is set and attempted to add non-internal user email
+	if !internalUser(useremail) && internal_fdi_bucket {
```

Comment:
Do you need a return statement in this "if", like the others? Or since this isn't a `if err != nil`, it does not need one?

Reply:
Since there should be no further action if this condition is met, then `return` seems to be correct